### ISSUE TITLE
Arrays can't be modified on the global scope when used as function arguments

### ISSUE THREAD

**Issue by marcosbitetti** [godotengine/godot#2108]
I try pass a array elements to a constructor like it

```
class GPointer:
    var cross
    var outline

    func _init(elements,t1,t2):
        cross = GRect.new(t1)
        outline = GRect.new(t2)
        elements.push_back(cross)
        elements.push_back(outline)

func _ready():
    var p = GPointer.new(elements,mira1,mira1_out)
    p.translate( Vector2(100,100) )
    pointers.push_back( p )
    update()

    set_process(true)
```

The result is blank array!!!!

The solution is work outside of class:

```
class GPointer:
    var cross
    var outline

    func _init(elements,t1,t2):
        cross = GRect.new(t1)
        outline = GRect.new(t2)
        elements.push_back(cross)
        #elements.push_back(outline)

func _ready():
    var p = GPointer.new(elements,mira1,mira1_out)
    p.translate( Vector2(100,100) )
    elements.push_back( p.cross )
    elements.push_back( p.outline )
    pointers.push_back( p )
    update()

    set_process(true)
```

This is no a first time that I have troubles with arrays in initializations https://github.com/okamstudio/godot/issues/2015

**Comment by slapin** [comment]
what are elements,mira1,mira1_out, how these are defined?
Godot is different than Python in the aspect that subclasses are isolated entities,
so don't know anything about class they are defined in.

So either there is something wrong with elements
or mira1, mira2, or with GRect. Probably that would be easier if you provide either debug info or full source code to check.

**Comment by marcosbitetti** [comment]
[grass_editor_ui_with_bug.zip](https://github.com/godotengine/godot/files/76037/grass_editor_ui_with_bug.zip)

Finally I reproduce this error in a project, appears that the complex initializations make it, then I compress entire project to show the bug.
See the comments on lines 273 and 275 from file grass_node.gd

**Comment by marcosbitetti** [comment]
Then I solve the problem:

    var rranges = []
    for i2 in range(layer_counters[i]):
    	msh.append(mesh_paths[index+i2])
    	if material_paths[index+i2].length()>0:
    		mat.append(material_paths[i])
    	else:
    		mat.append(null)
    	rranges.append( [RawArray(nears).get(index+i2),RawArray(fars).get(index+i2)] )
    layer_meshes.append(msh)
    layer_materials.append(mat)
    layers_metadata.append( {meshes=msh,materials=mat,ranges=rranges,randomization=randomizers[i]} )
    i += layer_counters[i]

**Comment by reduz** [comment]
I'm afraid neither I can understand the problem or make it crash :|

Does it still happen to you?

**Comment by marcosbitetti** [comment]
Yes :(
But after I make cast of Array to RawArray, it sync the array data I suppose .

**Comment by akien-mga** [comment]
First of all thank you for your report and sorry for the delay.

We released Godot 3.0 in January 2018 after 18 months of work, fixing many old issues either directly, or by obsoleting/replacing the features they were referring to.

We still have hundreds of issues whose relevance/reproducibility needs to be checked against the [current stable version](https://godotengine.org/download), and that's where you can help us.
**Could you check if the issue that you described initially is still relevant/reproducible in Godot 3.0 or any newer version, and comment about it here?**

For bug reports, please also make sure that the issue contains detailed steps to reproduce the bug and, if possible, a zipped project that can be used to reproduce it right away. This greatly speeds up debugging and bugfixing tasks for our contributors.

Our Bugsquad will review this issue more in-depth in 15 days, and potentially close it if its relevance could not be confirmed.

Thanks in advance.

*Note: This message is being copy-pasted to many "stale" issues (90+ days without activity). It might happen that it is not meaningful for this specific issue or appears oblivious of the issue's context, if so please comment to notify the Bugsquad about it.*

**Comment by akien-mga** [comment]
As there was no answer to the above request, we're closing this issue as inactive.
If you can still reproduce it, please open a new issue against a currently supported release of Godot.
